ambari-server/src/main/java/org/apache/ambari/server/serveraction/kerberos/KerberosOperationHandler.java

```diff
@@ -354,8 +354,9 @@ protected Keytab createKeytab(String principal, String password, Integer keyNumb
 
         keytab.setEntries(keytabEntries);
       }
+    } else {
+      throw new KerberosOperationException("Failed to create keytab file, the key encryption types you set are unsupported");
```

Comment:
i think that this is too late in the process to be useful.  However you might want to take a look at `org.apache.ambari.server.serveraction.kerberos.KerberosOperationHandler#translateEncryptionType` and throw an exception there if the requested encryption type is not valid.  

Or maybe do it in `org.apache.ambari.server.serveraction.kerberos.KerberosOperationHandler#translateEncryptionTypes` if none of the requested encryption types yield a supported type. 

I think you might want to log a message in `org.apache.ambari.server.serveraction.kerberos.KerberosOperationHandler#translateEncryptionType` and throw the exception in `org.apache.ambari.server.serveraction.kerberos.KerberosOperationHandler#translateEncryptionTypes`.

Reply:
Thanks you for your review!
Updated to log a message in `#translateEncryptionType` and throw the exception in `#translateEncryptionTypes`. Throwing the exception in `#translateEncryptionType` seems overkill.